Deploy Elastic Defend with MDM (#5749)
elastic/security-docs: docs/management/admin/ts-management.asciidoc

** `ping -4 -c 1 localhost` (Linux)
** `ping -c 1 localhost` (macOS)
** `ping -4 localhost` (Windows)
====

[discrete]
[[defend-deployment]]
.{elastic-defend} deployment issues
[%collapsible]
====

> {elastic-defend} deployments may result in warnings or errors in the endpoint **Policy status** in {fleet} due to incorrect mobile device management (MDM) configuration, or certain permissions not being enabled for {elastic-endpoint}. Here are some common issues that can cause the policy status to result in a warning or failure.

Reviewer: Switched around the order a bit and rewrote to remove some of the passive voice. 

```suggestion
After deploying {elastic-defend}, you might encounter warnings or errors in the endpoint **Policy status** in {fleet} if your mobile device management (MDM) is misconfigured or certain permissions for {elastic-endpoint} aren't granted. Read the following sections to learn more about issues that can cause warnings or failures in the endpoint policy status.
```

[float]

> ==== Connect Kernel has failed

Reviewer: Is this an actual warning or failure message that gets returned? If so, it might be worth clarifying in the intro for this section.

Author: It's not a verbatim error message—there would be a `Failed` status next to the **Connect Kernel** line in the UI.

> This means that the system extension or kernel extension was not approved. Consult the following topics for approving the system extension, either with MDM or without MDM:

Reviewer: ```suggestion
This means that the system extension or kernel extension was not approved. Consult the following topics for approving the system extension with or without MDM:
```

* <<system-extension-jamf, Approve the system extension with MDM>>
* <<system-extension-endpoint, Approve the system extension without MDM>>

You can validate the system extension is loaded by running
`sudo systemextensionsctl list | grep co.elastic.systemextension`. In the command output, the system extension should be marked as "active enabled".

[float]
==== Connect Kernel has failed and the system extension is loaded

> If the system extension is loaded and **Connect Kernel** still fails, this means that Full Disk Access was not granted. {elastic-endpoint} requires Full Disk Access to subscribe to system events via the {elastic-defend} framework, which is one of the primary sources of eventing information used by {elastic-endpoint}. Consult the following topics for granting Full Disk Access, either with MDM or without MDM:

Reviewer: ```suggestion
If the system extension is loaded and **Connect Kernel** still fails, this means that Full Disk Access was not granted. {elastic-endpoint} requires Full Disk Access to subscribe to system events through the {elastic-defend} framework, which is one of the primary sources of eventing information used by {elastic-endpoint}. Consult the following topics for granting Full Disk Access with or without MDM:
```

> If the system extension is loaded and **Connect Kernel** still fails, this means that Full Disk Access was not granted. {elastic-endpoint} requires Full Disk Access to subscribe to system events through the {elastic-defend} framework, which is one of the primary sources of eventing information used by {elastic-endpoint}. Consult the following topics for granting Full Disk Access with or without MDM:

Reviewer 2: Definitely not in Jamf fyi

> If the system extension is loaded and **Connect Kernel** still fails, this means that Full Disk Access was not granted. {elastic-endpoint} requires Full Disk Access to subscribe to system events through the {elastic-defend} framework, which is one of the primary sources of eventing information used by {elastic-endpoint}. Consult the following topics for granting Full Disk Access with or without MDM:

Author: It is a UI element in Elastic's UI, but I can use more general language (e.g. "...and kernel connection still fails..." ) if you think that's clearer.

> If the system extension is loaded and **Connect Kernel** still fails, this means that Full Disk Access was not granted. {elastic-endpoint} requires Full Disk Access to subscribe to system events through the {elastic-defend} framework, which is one of the primary sources of eventing information used by {elastic-endpoint}. Consult the following topics for granting Full Disk Access with or without MDM:

Reviewer 3: More general language would work, or extra context explaining where users would encounter the "Connect Kernel" element in the Elastic UI. I'm not familiar with it and we don't document it anywhere else in the Security docs, though that could just be one of our gaps.

* <<fda-jamf, Enable Full Disk Access with MDM>>
* <<enable-fda-endpoint-ven, Enable Full Disk Access without MDM>>

You can validate that Full Disk Access is approved by running
`sudo /Library/Elastic/Endpoint/elastic-endpoint test install`. If the command output doesn't contain a message about enabling Full Disk Access, the approval was successful.

[float]
==== Detect Network Events has failed

> This means that the network extension content filtering was not approved. Consult the following topics for approving network content filtering, either with MDM or without MDM:

Reviewer: ```suggestion
This means that the network extension content filtering was not approved. Consult the following topics for approving network content filtering with or without MDM:
```

* <<content-filtering-jamf, Approve network content filtering with MDM>>
* <<allow-filter-content-ven, Approve network content filtering without MDM>>

You can validate that network content filtering is approved by running
`sudo /Library/Elastic/Endpoint/elastic-endpoint test install`. If the command output doesn't contain a message about approving network content filtering, the approval was successful.

[float]
==== Full Disk Access has a warning

> This means that Full Disk Access was not granted for one or all {elastic-endpoint} components. Consult the following topics for granting Full Disk Access, either with MDM or without MDM:

Reviewer: ```suggestion
This means that Full Disk Access was not granted for one or all {elastic-endpoint} components. Consult the following topics for granting Full Disk Access with or without MDM:
```
